tests/helper/test_helper.py

```diff
@@ -603,28 +603,49 @@ def test_process_media_empty_url(
 
 
 @pytest.mark.parametrize(
-    "returncode, error",
+    "platform, returncode, tool, error",
     [
-        (0, []),
-        (
+        case("Linux", 0, "xdg-open", [], id="Linux:success"),
+        case("MacOS", 0, "open", [], id="MacOS:success"),
+        case("WSL", 1, "explorer.exe", [], id="WSL:success"),
+        case(
+            "Linux",
             1,
+            "xdg-open",
             [
                 " The tool ",
                 ("footer_contrast", "xdg-open"),
                 " did not run successfully" ". Exited with ",
                 ("footer_contrast", "1"),
             ],
+            id="Linux:error",
         ),
+        case(
+            "MacOS",
+            1,
+            "open",
+            [
+                " The tool ",
+                ("footer_contrast", "open"),
+                " did not run successfully" ". Exited with ",
+                ("footer_contrast", "1"),
```

Comment:
Reading this again, a reasonable PR after this would be to not output the error code, but provide more detailed output somewhere for followup later, a little like we do with the crash or threading errors. That, or try and diagnose further why the tool did not run properly.

We could also possibly provide a 'ran successfully' message in the footer for opening, but I've not looked at the code flow for that, and whether it would eg. overwrite other messages too rapidly?

Reply:
Understood 👍 

Wouldn't diagnosing why the tool did not run properly be platform dependent though?